ui/src/pages/version/HistoryDetails.js

```diff
@@ -14,48 +14,41 @@ import { useMerlinApi } from "../../hooks/useMerlinApi";
 
 const defaultTextSize = "s";
 
-const DeploymentStatus = ({
-  status,
-  deployment,
-  endpointRevisionId,
-  endpointStatus,
-}) => {
-  if (status === "running" || status === "serving") {
-    if (
-      deployment.revision_id === endpointRevisionId &&
-      (endpointStatus === "pending" ||
-        endpointStatus === "running" ||
-        endpointStatus === "serving")
-    ) {
+const DeploymentStatus = ({ status }) => {
+  switch (status) {
+    case "pending":
+      return <EuiHealth color="gray">Pending</EuiHealth>;
+    case "running":
+    case "serving":
       return <EuiHealth color="success">Deployed</EuiHealth>;
-    }
-    return <EuiHealth color="default">Not Deployed</EuiHealth>;
-  } else if (status === "pending") {
-    return <EuiHealth color="gray">Pending</EuiHealth>;
-  }
-
-  if (deployment.error !== "") {
-    return <EuiHealth color="danger">Failed</EuiHealth>;
+    case "terminated":
+      return <EuiHealth color="default">Not Deployed</EuiHealth>;
+    case "failed":
+      return <EuiHealth color="danger">Failed</EuiHealth>;
+    default:
+      return <EuiHealth color="subdued">-</EuiHealth>;
   }
 };
 
 const RevisionPanel = ({ deployments, deploymentsLoaded, endpoint }) => {
   const [orderedDeployments, setOrderedDeployments] = useState([]);
+  const [currentDeployment, setCurrentDeployment] = useState({ id: null });
+
   useEffect(() => {
-    let orderedDeployments = deployments.sort((a, b) => (a.id > b.id ? 1 : -1));
-
-    let idx = 0;
-    orderedDeployments.forEach((deployment) => {
-      if (deployment.status === "running" || deployment.status === "serving") {
-        deployment.revision_id = idx + 1;
-        idx++;
-      }
+    const ordered = deployments.sort((a, b) => (a.id < b.id ? 1 : -1));
+    setOrderedDeployments(ordered);
+
+    const currentDeployment = ordered.find((deployment) => {
+      return (
+        (deployment.status === "running" ||
```

Comment:
Should we just look for `deployment.status` != "failed"? Also, is there a reason we also need to check for `deployment.error === ""` besides the status?

Reply:
`deployment.status != "failed"` will include pending deployment too, however, we don't want to mark pending deployment as Current deployment.

We check `deployment.error === ""` for backward compatibility. Prior to Merlin 0.34 (model version revision introduction), deployments table might have the row with status in running and serving but with error message due to deployments data is always copying from the latest endpoint deployment's message. Ref: https://github.com/caraml-dev/merlin/blob/v0.33.0/api/queue/work/model_service_deployment.go#L95

Reply:
Got it. Thanks for the clarification!

Reply:
> deployments table might have the row with status in running and serving but with error message

Hmm.. Does it matter? If the deployment ended up to be "running" / "serving" despite the error, it should still be considered the current deployment, right?

Reply:
Oh, you're right. We don't need to check deployment.error value anymore if the status is already running or serving.